### ISSUE TITLE
Trees as first class citizens

### ISSUE THREAD

**Issue by talegari** [imbs-hl/ranger#490]
Hi Marvin,

I am trying to do these things:

1. Combine selected trees from multiple forests. Simplest case is combining two forests. (Partially addressed [here](https://github.com/imbs-hl/ranger/issues/356))

2. Prune a tree (or all or some trees in a forest) with cost complexity pruning or simply to a predefined depth.

3. Grow some existing trees to greater depth on newdata. This is an exercise as a part of continued learning on newer datasets. (handle drift in data as model phases out). ref: https://github.com/mlr-org/mlr3learners/issues/7

IMHO, this requires substantial new API on existing ranger class. Does this fit your vision of what ranger is supposed to do?

Regards,
Srikanth

**Comment by mnwright** [comment]
> IMHO, this requires substantial new API on existing ranger class.

I agree. We would need something like a _Tree_ class accessible from R. It might not only be a new API but a major re-design. 

> Does this fit your vision of what ranger is supposed to do?

I'm not sure. Currently, the trees are saved by three simple vectors (child nodes, split variables, split values). I like this simplicity. We might also run into problems keeping both R and C++ versions without too much maintenance overhead. 
On the other hand, as long as computation speed and memory is not affected, I'm not generally against a re-design.

**Comment by brshallo** [comment]
I'd be interested in item 2 as potentially useful in reviewing ways to reduce the size of ranger model objects, as discussed here: https://stackoverflow.com/questions/72961569/review-performance-of-smaller-model-subsets-of-a-large-random-forest-model?noredirect=1#comment128880412_72961569
